core/src/main/java/org/testcontainers/utility/CommandLine.java

```diff
@@ -57,15 +59,19 @@ public static String runShellCommand(String... command) {
      */
     public static boolean executableExists(String executable) {
 
+        String winExecutable = executable + ".exe";
+
         // First check if we've been given the full path already
         File directFile = new File(executable);
-        if (directFile.exists() && directFile.canExecute()) {
+        File windowsExeFile = new File(winExecutable);
```

Comment:
The clauses in the if statements below are beginning to get a little bit hairy! Perhaps instead we could have a conditional clause up here based on `isWindows()` to set the `File` and a path-`String` that can get used below. e.g.:
```
if (isWindows()) {
  fullExecutableName = executable + ".exe";
} else {
  fullExecutableName = executable;
}

directFile = new File(fullExecutableName);
```
Then we just need the tests in the `if` statements below to use `directFile` and `fullExecutableName` with less need for boolean operators.